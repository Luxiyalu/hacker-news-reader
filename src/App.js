import React from 'react';
import base from './config/base';
import Story from './Story/Story';
import './App.scss';

class App extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      stories: []
    };
  }

  componentDidMount() {
    base.bindToState('/v0/newstories', {
      state: 'stories',
      context: this
    });
  }

  render() {
    console.log('rerender:', this.state);

    return (
      <div className="App">
        {this.state.stories.slice(0, 20).map(id => (
          <Story key={id} id={id} />
        ))}
      </div>
    );
  }
}

export default App;
