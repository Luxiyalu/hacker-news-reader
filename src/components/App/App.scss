.hn-app {
  display: flex;
  justify-content: center;

  .center {
    max-width: 80%;
    display: flex;
    flex-grow: 1;
    align-content: center;
    flex-direction: column;
  }

  .stories-container {
    background-color: #f6f6ef;
  }
}

@media screen and (max-width: 940px) {
  .hn-app {
    .center {
      max-width: 100%;
    }
  }
}
