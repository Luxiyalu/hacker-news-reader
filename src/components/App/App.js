import React from 'react';
import Story from '../Story/Story';
import { base, storage } from '../../services/';
import './App.scss';

class App extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      limit: 20,
      stories: storage.getLatestStories() || []
    };
  }

  componentDidMount() {
    base.bindToState('/v0/newstories', {
      state: 'stories',
      context: this,
      then: () => {
        storage.setLatestStories(this.state.stories);
      }
    });
    window.addEventListener('scroll', this.handleScroll.bind(this));
  }

  componentWillUnmount() {
    window.removeEventListener('scroll', this.handleScroll);
  }

  handleScroll() {
    const { scrollTop, clientHeight, scrollHeight } = document.documentElement;

    if (scrollHeight - scrollTop - clientHeight < 100) {
      this.setState({ limit: this.state.limit + 20 });
    }
  }

  render() {
    console.log('rerender:', this.state);
    const { stories, limit } = this.state;

    return (
      <div className="hn-app">
        <header />

        <div className="stories-container">
          {stories.slice(0, limit).map(id => (
            <Story key={id} id={id} />
          ))}
        </div>

        <footer />
      </div>
    );
  }
}

export default App;
