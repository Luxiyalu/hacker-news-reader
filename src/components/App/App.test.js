import React from 'react';
import App from './App';
import renderer from 'react-test-renderer';

describe('<App />', () => {
  it('renders without crashing', () => {
    const app = renderer.create(<App />);
    const tree = app.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('initiate with the correct init state', () => {});

  it('calls preload on page load', () => {});

  it('increases display limit when scrolled to bottom', () => {});
});
