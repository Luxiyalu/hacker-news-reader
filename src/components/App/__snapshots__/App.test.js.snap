// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> renders without crashing 1`] = `
<div
  className="hn-app"
>
  <div
    className="center"
  >
    <header>
      <img
        alt="logo for HN"
        src="./logo.gif"
      />
      <h1>
        Hacker News
      </h1>
    </header>
    <div
      className="stories-container"
    />
  </div>
</div>
`;
